src/jshint.js

```diff
@@ -4186,6 +4215,15 @@ var JSHINT = (function() {
         error("W104", nextop, "for of");
       }
 
+      var ok = !(initializer || comma);
+      if (initializer) {
```

Comment:
Yeah, it wouldn't hurt to be opinionated about this. I was worried this would be a breaking change, but I just checked and JSHint doesn't allow this now (just with a less helpful error, "Expected 'in' and instead saw '='."). The only concern I have is that users cannot ignore errors. I don't know if defining an analogous warning for this problem is really worthwhile, though, because I can't think of any case where someone would intentionally want this.